src/lib/__tests__/InstantSearch-test.tsx

```diff
@@ -1289,9 +1289,16 @@ describe('scheduleRender', () => {
     expect(widget.render).toHaveBeenCalledTimes(0);
 
     search.on('render', () => {
-      expect(widget.render).toHaveBeenCalledTimes(1);
-      done();
+      renderSpy({
+        widgetRenderTimes: castToJestMock(widget.render!).mock.calls.length,
+      });
     });
+
+    await wait(0);
+
+    expect(renderSpy).toHaveBeenCalledTimes(2);
+    expect(renderSpy).toHaveBeenNthCalledWith(1, { widgetRenderTimes: 0 });
+    expect(renderSpy).toHaveBeenNthCalledWith(2, { widgetRenderTimes: 1 });
```

Comment:
This isn't obvious what the behavior is when just reading those lines, can you add comments?